src/plot_api/plot_api.js

```diff
@@ -19,345 +19,7 @@ var isNumeric = require('fast-isnumeric');
-                uid = trace.uid;
-                fullLayout._paper.selectAll('.hm'+uid+',.contour'+uid+',.cb'+uid+',#clip'+uid)
-                    .remove();
-            }
-        }
-
-        for (i = 0; i < subplots.length; i++) {
-            subplot = subplots[i];
-            subplotInfo = gd._fullLayout._plots[subplot];
-            cdSubplot = getCdSubplot(calcdata, subplot);
-            cdError = [];
-
-            // remove old traces, then redraw everything
-            // TODO: use enter/exit appropriately in the plot functions
-            // so we don't need this - should sometimes be a big speedup
-            if(subplotInfo.plot) subplotInfo.plot.selectAll('g.trace').remove();
-
-            for(j = 0; j < modules.length; j++) {
-                module = modules[j];
-                if(!module.plot) continue;
-
-                // plot all traces of this type on this subplot at once
-                cdModule = getCdModule(cdSubplot, module);
-                module.plot(gd, subplotInfo, cdModule);
-                Plotly.Lib.markTime('done ' + (cdModule[0] && cdModule[0][0].trace.type));
-
-                // collect the traces that may have error bars
-                if(cdModule[0] && cdModule[0][0].trace && plots.traceIs(cdModule[0][0].trace, 'errorBarsOK')) {
-                    cdError = cdError.concat(cdModule);
-                }
-            }
-
-            // finally do all error bars at once
-            if(gd._fullLayout._hasCartesian) {
-                Plotly.ErrorBars.plot(gd, subplotInfo, cdError);
-                Plotly.Lib.markTime('done ErrorBars');
-            }
-        }
-
-        // now draw stuff not on subplots (ie, pies)
-        // TODO: gotta be a better way to handle this
-        var cdPie = getCdModule(calcdata, Plotly.Pie);
-        if(cdPie.length) Plotly.Pie.plot(gd, cdPie);
-
-        // styling separate from drawing
-        plots.style(gd);
-        Plotly.Lib.markTime('done plots.style');
-
-        // show annotations and shapes
-        Plotly.Shapes.drawAll(gd);
-        Plotly.Annotations.drawAll(gd);
-
-        // source links
-        plots.addLinks(gd);
-
-        return plots.previousPromises(gd);
-    }
-
-    function cleanUp() {
-        // now we're REALLY TRULY done plotting...
-        // so mark it as done and let other procedures call a replot
-        gd._replotting = false;
-        Plotly.Lib.markTime('done plot');
-        gd.emit('plotly_afterplot');
-    }
-
-    var donePlotting = Plotly.Lib.syncOrAsync([
-        plots.previousPromises,
-        marginPushers,
-        layoutStyles,
-        marginPushersAgain,
-        positionAndAutorange,
-        drawAxes,
-        drawData
-    ], gd, cleanUp);
-
-    // even if everything we did was synchronous, return a promise
-    // so that the caller doesn't care which route we took
-    return (donePlotting && donePlotting.then) ?
-        donePlotting : Promise.resolve(gd);
-};
+Plotly.plot = require('./plot');
```

Comment:
My plan is to have `plot_api/index.js` that will export the plot_api parts and then I will bind them to Plotly in the upper level once that's done.

Ideally, we should follow a convention so that we *only* use the index file requires from higher-level files so that we can have at least some semblance of encapsulation and an easy way to trace code and functions back to their origins without too much trouble.

I think things like this are a bit of a code-smell.

```
var SomeModule = require('../somemodule');
var someFunction = require('../somemodule/somefunction');
```

It's a lot nicer in my mind, if instead we do:
```
var SomeModule = require('../somemodule');

// Somewhere down the file
SomeModule.someFunction();
```